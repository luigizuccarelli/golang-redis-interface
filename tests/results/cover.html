
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>handlers: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go (97.0%)</option>
				
				<option value="file1">gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package handlers

import (
        "bytes"
        "encoding/json"
        "fmt"
        "io/ioutil"
        "net/http"
        "os"

        "gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/connectors"
        "gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/schema"
        "github.com/microlib/simple"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

func LineDataHandler(w http.ResponseWriter, r *http.Request, logger *simple.Logger, con connectors.Clients) <span class="cov8" title="1">{
        var response *schema.Response
        var data []byte

        id := r.URL.Query().Get("id")
        addHeaders(w, r)

        if id == "" </span><span class="cov8" title="1">{
                id = "default-line"
        }</span>

        <span class="cov8" title="1">if r.Method == http.MethodGet </span><span class="cov8" title="1">{
                data, _ = con.Get(id + "-line")
        }</span> else<span class="cov8" title="1"> if r.Method == http.MethodPost </span><span class="cov8" title="1">{
                body, err := ioutil.ReadAll(r.Body)
                if err != nil </span><span class="cov8" title="1">{
                        response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "KO", Message: fmt.Sprintf("Could not read body data %v\n", err), Payload: ""}
                        w.WriteHeader(http.StatusInternalServerError)
                }</span> else<span class="cov8" title="1"> {
                        con.Set(id+"-line", body, 0)
                        data = body
                }</span>
        }
        <span class="cov8" title="1">response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "200", Status: "OK", Message: "Data processed succesfully", Payload: string(data)}
        w.WriteHeader(http.StatusOK)
        b, _ := json.MarshalIndent(response, "", "        ")
        logger.Debug(fmt.Sprintf("LineDataHandler response : %s", string(b)))
        fmt.Fprintf(w, string(b))</span>
}

func BarDataHandler(w http.ResponseWriter, r *http.Request, logger *simple.Logger, con connectors.Clients) <span class="cov8" title="1">{
        var response *schema.Response
        var data []byte

        id := r.URL.Query().Get("id")
        addHeaders(w, r)

        if id == "" </span><span class="cov8" title="1">{
                id = "default-bar"
        }</span>

        <span class="cov8" title="1">if r.Method == http.MethodGet </span><span class="cov8" title="1">{
                data, _ = con.Get(id + "-bar")
        }</span> else<span class="cov8" title="1"> if r.Method == http.MethodPost </span><span class="cov8" title="1">{
                body, err := ioutil.ReadAll(r.Body)
                if err != nil </span><span class="cov8" title="1">{
                        response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "KO", Message: fmt.Sprintf("Could not read body data %v\n", err), Payload: ""}
                        w.WriteHeader(http.StatusInternalServerError)
                }</span> else<span class="cov8" title="1"> {
                        con.Set(id+"-bar", body, 0)
                        data = body
                }</span>
        }
        <span class="cov8" title="1">response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "200", Status: "OK", Message: "Data processed succesfully", Payload: string(data)}
        w.WriteHeader(http.StatusOK)
        b, _ := json.MarshalIndent(response, "", "        ")
        logger.Debug(fmt.Sprintf("BarDataHandler response : %s", string(b)))
        fmt.Fprintf(w, string(b))</span>
}

func ApiCallHandler(w http.ResponseWriter, r *http.Request, logger *simple.Logger, con connectors.Clients) <span class="cov8" title="1">{
        var response *schema.Response

        body, err := ioutil.ReadAll(r.Body)
        if err != nil </span><span class="cov8" title="1">{
                response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "KO", Message: fmt.Sprintf("Could not read body data %v\n", err), Payload: ""}
                w.WriteHeader(http.StatusInternalServerError)
        }</span>
        <span class="cov8" title="1">req, err := http.NewRequest("POST", os.Getenv("URL"), bytes.NewBuffer(body))
        resp, err := con.Do(req)
        defer resp.Body.Close()
        if err != nil || resp.StatusCode != 200 </span><span class="cov8" title="1">{
                logger.Error(fmt.Sprintf("Http request %v\n", err))
                response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "KO", Message: fmt.Sprintf("Http request error %v\n", err), Payload: ""}
                w.WriteHeader(http.StatusInternalServerError)
        }</span> else<span class="cov8" title="1"> {
                body, err := ioutil.ReadAll(resp.Body)
                if err != nil </span><span class="cov0" title="0">{
                        response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "KO", Message: fmt.Sprintf("Could not read body data %v\n", err), Payload: ""}
                        w.WriteHeader(http.StatusInternalServerError)
                }</span> else<span class="cov8" title="1"> {
                        response = &amp;schema.Response{Name: os.Getenv("NAME"), StatusCode: "200", Status: "OK", Message: "ApiCall request processed succesfully", Payload: string(body)}
                        w.WriteHeader(http.StatusOK)
                }</span>
        }
        <span class="cov8" title="1">b, _ := json.MarshalIndent(response, "", "        ")
        logger.Debug(fmt.Sprintf("ApiCallHandler response : %s", string(b)))
        fmt.Fprintf(w, string(b))</span>
}

func IsAlive(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        fmt.Fprintf(w, "{ \"version\" : \"1.0.2\" , \"name\": \""+os.Getenv("NAME")+"\" }")
}</span>

// headers (with cors) utility
func addHeaders(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        w.Header().Set(CONTENTTYPE, APPLICATIONJSON)
        // use this for cors
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
        w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
}</span>
</pre>
		
		<pre class="file" id="file1" style="display: none">package validator

import (
        "errors"
        "fmt"
        "os"
        "strconv"
        "strings"

        "github.com/microlib/simple"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string, logger *simple.Logger) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        logger.Trace(fmt.Sprintf("name %s : required %t", name, required))
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Error(fmt.Sprintf("%s envar is mandatory please set it", name))
                        return errors.New(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span> else<span class="cov8" title="1"> {
                        logger.Error(fmt.Sprintf("%s envar is empty please set it", name))
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
// Each microservice will obviously have a diffefrent envars so change where needed
func ValidateEnvars(logger *simple.Logger) error <span class="cov8" title="1">{
        items := []string{
                "LOG_LEVEL,false",
                "SERVER_PORT,true",
                "REDIS_HOST,true",
                "REDIS_PORT,true",
                "REDIS_PASSWORD,true",
                "VERSION,true",
                "URL,true",
        }
        for x, _ := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x], logger); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
