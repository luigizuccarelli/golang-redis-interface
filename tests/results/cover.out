mode: set
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:14.59,18.27 4 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:26.2,26.12 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:18.27,19.15 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:19.15,22.4 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:22.9,24.4 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:32.50,42.26 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:47.2,47.12 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:42.26,43.54 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/validator/validate.go:43.54,45.4 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:21.110,28.14 6 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:31.2,31.17 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:35.2,35.32 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:47.2,51.27 5 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:28.14,30.3 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:31.17,33.3 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:35.32,37.3 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:37.8,37.40 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:37.40,39.17 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:39.17,42.4 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:42.9,45.4 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:54.108,58.16 3 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:62.2,65.42 4 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:79.2,81.27 3 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:58.16,61.3 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:65.42,69.3 3 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:69.8,71.17 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:71.17,74.4 2 0
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:74.9,77.4 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:84.54,86.2 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/golang-redisapi/pkg/handlers/handlers.go:89.57,95.2 4 1
